llvm/lib/Target/NVPTX/NVPTXISelDAGToDAG.cpp

```diff
@@ -445,6 +448,17 @@ bool NVPTXDAGToDAGISel::SelectSETP_BF16X2(SDNode *N) {
   return true;
 }
 
+bool NVPTXDAGToDAGISel::tryUNPACK_VECTOR(SDNode *N) {
+  SDValue Vector = N->getOperand(0);
+  MVT EltVT = N->getSimpleValueType(0);
+
+  MachineSDNode *N2 =
+      CurDAG->getMachineNode(NVPTX::I64toV2I32, SDLoc(N), EltVT, EltVT, Vector);
+
+  ReplaceNode(N, N2);
+  return true;
+}
+
```

Comment:
Why can't this be done in `NVPTXInstrInfo.td`?

Reply:
I tried this but wasn't able to do it without llvm-tblgen crashing or hanging. I think table-gen support for nodes with multiple return values is not very reliable.